registry/handlers/images.go

```diff
@@ -179,8 +172,7 @@ func (imh *imageManifestHandler) DeleteImageManifest(w http.ResponseWriter, r *h
 	// tag index entries a serious problem in eventually consistent storage.
 	// Once we work out schema version 2, the full deletion system will be
 	// worked out and we can add support back.
-	imh.Errors.Push(v2.ErrorCodeUnsupported)
-	w.WriteHeader(http.StatusBadRequest)
+	imh.Errors = append(imh.Errors, errcode.NewError(v2.ErrorCodeUnsupported))
```

Comment:
This is a great example of where we should have `ErrorCode` implement `error. There is no reason to allocate an error here.